workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java

```diff
@@ -194,6 +200,38 @@ public WorkFlowStatusResponseDTO getWorkFlowStatus(UUID workFlowExecutionId) {
 				.works(workFlowWorksStatusResponseDTOs).build();
 	}
 
+	@Override
+	public WorkFlowContextResponseDTO getWorkflowParameters(UUID workFlowExecutionId,
+			List<WorkContextDelegate.Resource> params) {
+		WorkFlowExecution workFlowExecution = workFlowRepository.findById(workFlowExecutionId).orElseThrow(() -> {
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND,
+					String.format("workflow execution id: %s not found!", workFlowExecutionId));
+		});
+		Map options = null;
+		if (params.contains(WorkContextDelegate.Resource.WORKFLOW_OPTIONS)) {
+			options = (Map) WorkContextDelegate.read(workFlowExecution.getWorkFlowExecutionContext().getWorkContext(),
+					WorkContextDelegate.ProcessType.WORKFLOW_EXECUTION, WorkContextDelegate.Resource.WORKFLOW_OPTIONS);
+		}
+		if (options == null) {
```

Comment:
init options with `Map.of`

Reply:
can we define options as WorkFlowOptions as we did before,  and use object mapper to convert 
`WorkContextDelegate.read(workFlowExecution.getWorkFlowExecutionContext().getWorkContext(),
					WorkContextDelegate.ProcessType.WORKFLOW_EXECUTION, WorkContextDelegate.Resource.WORKFLOW_OPTIONS)` to it?

Reply:
I'm not sure this is the same as before, since when we used to retrieve the `WorkFlowOptions` as part of the `execute()` API, the context remained in memory and wasn't read from the DB.
That means that the actual type of `workFlowExecution.getWorkFlowExecutionContext().getWorkContext(), WorkContextDelegate.ProcessType.WORKFLOW_EXECUTION, WorkContextDelegate.Resource.WORKFLOW_OPTIONS)`  was `WorkflowOptions` as set to it in the workflow-examples.

With this change, the value is being read from the DB, which should convert the json representation of the workflow-execution-context to a WorkFlowOptions.

I don't like the fact we might have more than one type for the same object being held in the context map.
Perhaps this should be solved on the DB layer to convert that part of context from a map to WorkFlowOptions.

Do you have anything specific in mind about how to treat this one?

Reply:
my thinking of using WorkFlowOptions is that dto is more structured than Map.  But you are right that from the perspective of whole workcontext,  it's better to keep the value type as `Map`. Maybe in future we can specify the Type for specific Resources(e.g. WORKFLOW_OTPIONS -> WorkFlowOptions).  Let's keep `Map` for now